lib/model.py

```diff
@@ -0,0 +1,180 @@
+import abc
+
+import numpy as np
+import torch
+from alibi_detect.cd import (ChiSquareDrift, ClassifierDrift,
+                             ClassifierUncertaintyDrift, CVMDrift, FETDrift,
+                             KSDrift, LearnedKernelDrift, LSDDDrift, MMDDrift,
+                             SpotTheDiffDrift, TabularDrift)
+from alibi_detect.utils.pytorch.kernels import DeepKernel
+from sklearn.ensemble import RandomForestClassifier
+from sklearn.metrics import f1_score
+from torch import nn
+
+# TODO: implement preprocessing for feature reduction in all models
+class ModelWrapperInterface(abc.ABC):
+    '''
+        Drift detector wrapper.
+    '''
+    @abc.abstractmethod
+    def fit(self, x_fit, y_fit):
+        '''
+            Initializes a drift detector with some combination of
+            x_fit, y_fit. x_fit, y_fit can be of any type, depending
+            on the data loader implementation.
+        '''
+        pass
+    @abc.abstractmethod
+    def infer(self, x_infer, y_infer) -> float:
+        '''
+            Returns p-value corresponding to drift for the drift
+            detector initialized in fit method.
+        '''
+        pass
+
+@ModelWrapperInterface.register
+class CVMModel:
```

Comment:
Instead of using the register decorator, you can also just do:

```python
class MMDModel(ModelWrapperInterface):
...
```

Reply:
Done. This makes more sense since the register decorator seems to be used when the implementation isn't in the inheritance tree of the abc.